### DISCUSSION TITLE
Turning a lamp on with transition does not seem to work

### DISCUSSION THREAD

**Discussion by kszys** [Koenkk/zigbee2mqtt#19075] [Devices]
I run into some issues when activating Home Assistant Transitions on zigbee2mqtt-connected Ikea bulbs. The MQTT message from HA seems OK, but then I do not see any log of any transition being initiated on the zigbee2mqtt side:

```
Debug 2023-09-25 14:02:39Received MQTT message on 'zigbee2mqtt/Office Lamp/set' with data '{"state":"ON","transition":300.0,"brightness":127,"color_temp":357}'
Debug 2023-09-25 14:02:39Publishing 'set' 'brightness' to 'Office Lamp'
Debug 2023-09-25 14:02:39Publishing 'set' 'transition' to 'Office Lamp'
Debug 2023-09-25 14:02:39Publishing 'set' 'color_temp' to 'Office Lamp'
Debug 2023-09-25 14:02:39Received Zigbee message from 'Office Lamp', type 'attributeReport', cluster 'genOnOff', data '{"onOff":1}' from endpoint 1 with groupID 0
Info 2023-09-25 14:02:39MQTT publish: topic 'zigbee2mqtt/Office Light', payload '{"brightness":127,"color_mode":"color_temp","color_temp":357,"state":"ON"}'
Info 2023-09-25 14:02:39MQTT publish: topic 'zigbee2mqtt/Office Lamp', payload '{"brightness":127,"color_mode":"color_temp","color_temp":357,"level_config":{"on_level":"previous"},"linkquality":204,"power_on_behavior":null,"state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null}'
Info 2023-09-25 14:02:39MQTT publish: topic 'zigbee2mqtt/Office Light', payload '{"brightness":127,"color_mode":"color_temp","color_temp":357,"state":"ON"}'
Info 2023-09-25 14:02:39MQTT publish: topic 'zigbee2mqtt/Office Lamp', payload '{"brightness":127,"color_mode":"color_temp","color_temp":357,"level_config":{"on_level":"previous"},"linkquality":204,"power_on_behavior":null,"state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null}'
```

You can see the first message indicating that message `zigbee2mqtt/Office Lamp/set` came with data: 

```
{"state":"ON","transition":300.0,"brightness":127,"color_temp":357}
```

However, while one of the following lines says:

```
Debug 2023-09-25 14:02:39Publishing 'set' 'transition' to 'Office Lamp'
```

in fact none of the publisheded payloads seem to indicate any transition being sent to the lamp...

**Comment by uvok** [comment]
Hey! 1 year late, but I had the same problem with z2m 1.40.2 and an Ikea LED2003G10

The problem seems to be using e.g. `{"state":"ON","transition":10, "color_temp": 400}` doesn't work - when I leave out the color_temp, the light switches on with transition!

**Reply by Koenkk** [reply]
IKEA doesn't support transitions on both brightness and color temp at once, therefore the color temp transition is ignored when also changing the brightness (which happens when turning the bulb on)

**Reply by uvok** [reply]
Good to know. Should this be documented, e.g. in https://www.zigbee2mqtt.io/devices/LED2003G10.html#transition ?

Also, the behavior I was observing: When sending the command above, the light turns on immediately, and then seems to do the color transition.

**Reply by Koenkk** [reply]
Done!

**Comment by Stooovie** [comment]
The [bulb](https://www.zigbee2mqtt.io/devices/LED2003G10.html#ikea-led2003g10) turns on without transition even when only Brightness is specified. I don't use Color Transitions. The proper transition is present when controlling the bulb from its Z2M UI page, but not from HA.

EDIT: sorry, it was Adaptive Lighting interfering, injecting its own color temp settings. "Intercept" has to be disabled in it in order to keep the transitions. Leaving this here for posterity.
